EndpointCategory/helpers/instantiateEndpoint.js

```diff
@@ -87,84 +87,114 @@ module.exports = (config) => {
-            zoomErrorMessage = body.message;
+            if (typeof body.message === 'string') {
+              zoomErrorMessage = body.message;
+            }
           }
-        }
-        // Note: if none of the conditionals hit,
-        // zoomErrorMessage defaults to generic error message
+          // Note: if none of the conditionals hit,
+          // zoomErrorMessage defaults to generic error message
 
-        const errorMessage = `We couldn't ${action} because an error occurred: ${zoomErrorMessage}`;
-        const errorCode = `ZOOM${status}${body.code ? `-${body.code}` : ''}`;
+          const errorMessage = `We couldn't ${action} because an error occurred: ${zoomErrorMessage}`;
+          const errorCode = `ZOOM${status}${body.code ? `-${body.code}` : ''}`;
 
-        throw new ZACCLError({
-          message: errorMessage,
-          code: errorCode,
-        });
-      }
+          throw new ZACCLError({
+            message: errorMessage,
+            code: errorCode,
+          });
+        }
 
-      // Run the post-processor, throwing any errors it produces
-      // and convert non-ZACCLError errors into ZACCLErrors with better text
-      let modifiedResponse = response;
-      if (postProcessor) {
-        try {
-          modifiedResponse = postProcessor(response);
-        } catch (err) {
+        // Run the post-processor, throwing any errors it produces
+        // and convert non-ZACCLError errors into ZACCLErrors with better text
+        let modifiedResponse = response;
+        if (postProcessor) {
+          try {
+            modifiedResponse = postProcessor(response);
+          } catch (err) {
           // Turn into ZACCLError if not already
-          let newError = err;
-          if (!err.isZACCLError) {
-            newError = new ZACCLError(err);
-            newError.message = 'An error occurred while post-processing the response from Zoom.';
-            newError.code = ERROR_CODES.POST_PROCESSING_ERROR;
+            let newError = err;
+            if (!err.isZACCLError) {
+              newError = new ZACCLError(err);
+              newError.message = 'An error occurred while post-processing the response from Zoom.';
+              newError.code = ERROR_CODES.POST_PROCESSING_ERROR;
+            }
+            throw newError;
           }
-          throw newError;
         }
-      }
-      // Return the body of the zoom response
-      return modifiedResponse;
+
+        // Response is valid. Add response to pages
+        pages.push(modifiedResponse);
+
+        // Check for next page
+        if (modifiedResponse.body.next_page_token) {
+          // Add next page token to the params
+          endpointOptions.params
+            .next_page_token = modifiedResponse.body.next_page_token;
+
+          // Fetch next page recursively
+          return fetchPage();
+        }
+
+        // We don't need to fetch any more pages.
+        // Concatenate pages if necessary
```

Comment:
Isn't pages an array of objects which themselves (potentially) contain an array of results? I assume it's those arrays which we want to concatenate, not the pages array itself, which should already be flat, although I'm not sure how to do so generically across all endpoints. Additionally, methods which return arrays (such as List Recordings) also return some pagination metadata, which we might want to modify in some way. This might need to be handled individually per endpoint.

Reply:
Each page should be an array *after the postProcessor runs*

Reply:
(So I think Grace is right _and_ we need to make sure the `postProcessor` runs on each page before concatenation)

Reply:
Ah right, forgot about the postProcessor

Reply:
Yup, we're pushing after the postProcessor does its magic